mysql/resource_user.go

```diff
@@ -121,14 +156,23 @@ func CreateUser(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 
 	if authStm != "" {
 		stmtSQL = stmtSQL + authStm
-	} else {
+	} else if password != "" {
 		stmtSQL = stmtSQL + fmt.Sprintf(" IDENTIFIED BY '%s'", password)
 	}
 
 	requiredVersion, _ := version.NewVersion("5.7.0")
 
+	var updateStmtSql = ""
+
 	if getVersionFromMeta(ctx, meta).GreaterThan(requiredVersion) && d.Get("tls_option").(string) != "" {
-		stmtSQL += fmt.Sprintf(" REQUIRE %s", d.Get("tls_option").(string))
+		if createObj == "AADUSER" {
```

Comment:
I don't understand this part - doesn't `CREATE AADUSER` allow `REQUIRE ...`? Is there any documentation of the grammar?

It has also an undesired effect: when ALTER fails, the user will exist, but terraform won't know.

Reply:
aad_auth is something not-well documented by Microsoft now as it's just public preview feature - [this page](https://learn.microsoft.com/en-us/azure/mysql/flexible-server/how-to-azure-ad#create-azure-ad-users-in-azure-database-for-mysql) and my own trial and error...

So yes, `CREATE AADUSER` does not support `REQUIRE...`  and it's reason why I split it into two.

Will send update which handles partially created user in a minute. Thanks for pointing that out.

Reply:
Line 214 (`d.Set("tls_option", "")`) does the trick and terraform manages state of partially created user if `ALTER` fails